electron/process_manager.ts

```diff
@@ -146,27 +152,36 @@ function startLocalShadowsocksProxy(
     ssLocalArgs.push('-m', serverConfig.method || '');
     ssLocalArgs.push('-u');
 
-    try {
-      ssLocal = spawn(pathToEmbeddedExe('ss-local'), ssLocalArgs, {stdio: 'ignore'});
+    // Note that if you run with -v then ss-local may output a lot of data to stderr which
+    // will cause the binary to fail:
+    //   https://nodejs.org/dist/latest-v10.x/docs/api/child_process.html#child_process_maxbuffer_and_unicode
+    ssLocal = spawn(pathToEmbeddedExe('ss-local'), ssLocalArgs);
 
-      ssLocal.on('exit', (code, signal) => {
-        // We assume any signal sent to ss-local was sent by us.
-        if (signal) {
-          sentryLogger.info(`ss-local exited with signal ${signal}`);
-          onDisconnected();
-          return;
-        }
+    // Amazingly, there's no documented way to tell whether spawn has successfully launched a
```

Comment:
I don't think that's possible. The process starts immediately, the rest is application-specific. You need some sort of communication to tell the application is running properly. In this case it's the stdout output.

Reply:
Okay, perhaps I'm being too hard on `spawn` :-)